src/theory/bv/theory_bv_rewrite_rules_simplification.h

```diff
@@ -757,7 +757,9 @@ bool RewriteRule<MultPow2>::applies(TNode node) {
     return false;
 
   for(unsigned i = 0; i < node.getNumChildren(); ++i) {
```

Comment:
Since you already touch this code, can you maybe get rid of the call to getNumChildren() in each loop iteration?